kernel/pipedev.c

```diff
@@ -737,25 +737,39 @@ static int _pd_ioctl(
         arg,
         myst_getppid()));
 
-    if (request == TIOCGWINSZ)
+    switch (request)
     {
-        ERAISE(-EINVAL);
-    }
-    else if (request == FIONBIO)
-    {
-        int* val = (int*)arg;
-
-        if (!val)
+        case TIOCGWINSZ:
+        {
             ERAISE(-EINVAL);
+            break;
+        }
+        case FIONBIO:
+        {
+            int* val = (int*)arg;
 
-        if (*val)
-            pipe->fl_flags |= O_NONBLOCK;
-        else
-            pipe->fl_flags &= ~O_NONBLOCK;
-    }
-    else
-    {
-        ERAISE(-ENOTSUP);
+            if (!val)
+                ERAISE(-EINVAL);
+
+            if (*val)
+                pipe->fl_flags |= O_NONBLOCK;
```

Comment:
`pipedev.c` has `#define FL_FLAGS (O_APPEND | O_ASYNC | O_DIRECT | O_NOATIME | O_NONBLOCK)`, which is used for `fnctl`. Is `O_NONBLOCK` the only applicable flag here.

Reply:
Currently, Mystikos only partially supports ioctl (for all supported dev types). Further enhancement will be evaluated and prioritized based on targeted programing language runtime needs.